### PR TITLE
Add more type hints, fix some pylint warnings

### LINKED ISSUE #66: Move pylint exclude to pyproject
Let's move pylint exclude from precommits to pyproject.toml

See https://github.com/leikoilja/glocaltokens/pull/56#discussion_r608507433

### PR DESCRIPTION
**Pull request by KapJI**
Mypy helps to catch places where `Optional` is not handled properly.

Fixes #66

### CODE REVIEW

File: glocaltokens/client.py [hunk 1]
```diff
@@ -289,20 +288,26 @@ def get_homegraph(self):
                 "There is no stored homegraph, or it has expired, getting a new one..."
             )
             log_prefix = "[GRPC]"
+            access_token = self.get_access_token()
+            if not access_token:
+                LOGGER.debug("%s Unable to obtain access token.", log_prefix)
+                return None
```
Comment (leikoilja) [inline glocaltokens/client.py:294]: I'm wondering if we should extract that check into a helper function decorator or just a helper function to call at all places as we do now

File: pyproject.toml [hunk 2]
```diff
@@ -51,12 +51,21 @@ ipdb = "^0.13.7"
 extension-pkg-whitelist = [
     "_socket",
 ]
+ignore = ["google"]
```
Comment (leikoilja) [inline pyproject.toml:54]: nice!

File: pyproject.toml [hunk 3]
```diff
@@ -51,12 +51,21 @@ ipdb = "^0.13.7"
...
+
+[tool.pylint.basic]
+good-names = [
+    "zc",
+]
```
Comment (leikoilja) [inline pyproject.toml:59]: What is that for? 🤔

File: tests/test_client.py [hunk 4]
```diff
@@ -250,7 +251,7 @@ def test_get_access_token(self, m_perform_oauth, m_get_master_token, m_log):
     @patch("glocaltokens.client.GLocalAuthenticationTokens.get_access_token")
     def test_get_homegraph(
         self,
-        m_get_access_token,  # pylint: disable=unused-argument
+        _m_get_access_token,
```
Comment (leikoilja) [inline tests/test_client.py:254]: uuuu, didn't know you can do that 👍

File: tests/test_client.py [hunk 5]
```diff
@@ -279,6 +280,7 @@ def test_get_homegraph(
         self.assertEqual(m_get_home_graph_request.call_count, 1)
 
         # Expired homegraph
+        assert self.client.homegraph_date is not None
```
Comment (leikoilja) [inline tests/test_client.py:283]: Let's have here `unittest` pattern, just to have it consistent with the rest of the code?
```suggestion
        self.assertIsNotNone(self.client.homegraph_date)
```

**Review by leikoilja — APPROVED** [review]
Thanks, those are some really good changes 💯

File: pyproject.toml (hunk 3 above)
Comment (KapJI) [reply pyproject.toml:59]: To fix `invalid-name` pylint warning to `ServiceListener` methods.

File: tests/test_client.py (hunk 5 above)
Comment (KapJI) [reply tests/test_client.py:283]: Unfortunately, `mypy` can't narrow type from such calls: https://github.com/python/mypy/issues/4063

**Pull request merged**